src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -115,6 +114,9 @@ public class SelectStatement implements CQLStatement.SingleKeyspaceCqlStatement
 
     private final boolean isReversed;
 
+    private static int columnIndex;
+    private static SingleRestriction restriction;
```

Comment:
These are declared, but never set. Where should we set them? Also, we likely don't want them to be `static` fields since they will be different for each `SelectStatement` instance.

Reply:
Most of the vector related tests are failing because the `restriction` is null.

```
java.lang.NullPointerException
	at org.apache.cassandra.cql3.statements.SelectStatement.orderResults(SelectStatement.java:1027)
	at org.apache.cassandra.cql3.statements.SelectStatement.process(SelectStatement.java:904)
	at org.apache.cassandra.cql3.statements.SelectStatement.processResults(SelectStatement.java:520)
	at org.apache.cassandra.cql3.statements.SelectStatement.executeInternal(SelectStatement.java:544)
	at org.apache.cassandra.cql3.statements.SelectStatement.executeLocally(SelectStatement.java:526)
	at org.apache.cassandra.cql3.statements.SelectStatement.executeLocally(SelectStatement.java:101)
	at org.apache.cassandra.cql3.QueryProcessor.executeInternal(QueryProcessor.java:483)
	at org.apache.cassandra.cql3.CQLTester.executeFormattedQuery(CQLTester.java:1368)
	at org.apache.cassandra.cql3.CQLTester.execute(CQLTester.java:1356)
	at org.apache.cassandra.index.sai.cql.VectorAndLuceneTest.partitionDeleteVectorInMemoryTest(VectorAndLuceneTest.java:86)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitForkedStarter.main(JUnitForkedStarter.java:29)
```

Reply:
I removed them and the tests seem to be passing now, as per the refactored `orderResults` method at [this commit](https://github.com/datastax/cassandra/pull/710/commits/22cf12dc806b99372f602ec7f7aff702fb727191).